galsim/integ.py

```diff
@@ -154,3 +154,54 @@ def calculateWaves(self, bandpass):
             return [bandpass.blue_limit + h * i for i in range(self.N+1)]
         else:
             return [bandpass.blue_limit + h * (i+0.5) for i in range(self.N)]
+
+class DirectImageIntegrator(object):
+    def __init__(self, rule):
+        raise NotImplementedError("Must instantiate subclass of DirectImageIntegrator")
+    # subclasses must define
+    # 1) a method `.calculateWaves(bandpass)` which will return the wavelengths at which to
+    #    evaluate the integrand
+    # 2) an function attribute `.rule` which takes a list of integrand evaluations as its first
+    #    argument, and a list of evaluation wavelengths as its second argument, and returns
+    #    an approximation to the integral.  (E.g., the function midpt above, or numpy.trapz)
+    def __call__(self, imageAtWavelength, bandpass):
+        """
+        @param imageAtWavelength    Function that returns an Image of the monochromatic surface
+                                    brightness profile as a function of wavelength.
+        @param bandpass             Bandpass object representing the filter being imaged through.
+
+        @returns the result of integral as an Image
+        """
+        images = []
+        stepk = []
+        maxk = []
+        waves = self.calculateWaves(bandpass)
+        self.last_n_eval = len(waves)
+        for w in waves:
+            ret_val, tmp_stepk, tmp_maxk = imageAtWavelength(w)
+            images.append(bandpass(w)*ret_val)
+            stepk.append(tmp_stepk)
+            maxk.append(tmp_maxk)
+        return self.rule(images, waves), min(stepk), max(maxk)
+
+class DirectSampleIntegrator(DirectImageIntegrator):
```

Comment:
Yes, it's a relic.  I removed it; we can always restore it later if we decide we need it for something.